interview-questions.md

```diff
@@ -8,42 +8,55 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 2. What is the difference between a Float and an Integer in Ruby?
 
-Your answer:
+Your answer: Floats and integers are both numbers, and in Ruby, Float and Integer are two different classes. Floats are numbers with decimals, and even what what we could consider a whole number like 8.0 would be considered a float because there is a decimal point. Meanwhile, integers are just whole numbers. Both can be used to perform arithmetic and are used in methods.
 
-Researched answer:
+Researched answer: In Ruby, numbers can be classified as an Integer or a Float. Integers are whole numbers that can have a positive, negative, or zero value. Floats are numbers that contain a fractional part, so any number with a decimal. When we perform arithmetic with these two classes, performing operations with two integers will result in an integer. And if a float is involved in the arithmetic, the result will be a float.
 
 3. Ruby has an implicit return. What does that mean?
 
-Your answer:
+Your answer: An implicit return is a one that does not require the keyword "return". This is how Ruby returns methods. Every method in Ruby starts with a defining keyword and ends in "end". Within this scope of the method, the last line of code before end is always the return.
 
-Researched answer:
+Researched answer: Ruby automatically returns the last line of code in a block, so it is not necessary to write the "return" keyword to receive a return. It is implied that the last line of code that is executed will be the return.
+
+def add_two num
+    num + 2
+end
+
+In this method, there is no return keyword, but Ruby knows to return num + 2 as it is the last line of code in the method to be executed.
 
 4. What is a block in Ruby?
 
-Your answer:
+Your answer: A block in Ruby contains code that that can be passed onto a method to determine what that method does. It is an anonymous function that typically starts with "do" and ends with "end". Within the do/end, we can create behavior that passes through the method rather than just data. Whatever code is written within the code determines the scope, so any variables or data written within the block is not accessible outside of it.
+
+Researched answer: A block in Ruby is an anonymous function. It does not have a name but is used to manipulate data in methods with behavior. They contain chunks of code that can take an argument and return a value. In Ruby, the scope of a block is determined with the keywords "do" and "end". The parameters passed through blocks are written inside pipes ||. When using enumerables, we typically use blocks to write the behavior that we want to execute.
+
+num_array = [2, 3, 4, 5]
 
-Researched answer:
+num_array.map do |num|
+    num + 2
+end
+==> [4, 5, 6, 7]
 
```

Comment:
nice. Your knowledge really shines with this approach. Recommend you maintain this flow.